File: 실습/5장/04_lambda_collection/Main.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Main {
    static class Person {
        private final String name;
        private final int age;

        Person(String name, int age) {
            this.name = name;
            this.age = age;
        }

        public String getName() {
            return name;
        }

        public int getAge() {
            return age;
        }
    }

    public static void main(String[] args) {
        List<Person> people = Arrays.asList(
                new Person("철수", 17),
                new Person("영희", 20),
                new Person("민수", 25)
        );

        List<String> adultNames = people.stream()
                .filter(person -> person.getAge() >= 20)
                .map(person -> person.getName())
                .collect(Collectors.toList());

        System.out.println(adultNames);
    }
}

File: 실습/5장/04_lambda_collection/Main.kt
data class Person(
    val name: String,
    val age: Int
)

fun main() {
    val people = listOf(
        Person("철수", 17),
        Person("영희", 20),
        Person("민수", 25)
    )

    val adultNames = people
        .filter { it.age >= 20 }
        .map { it.name }

    println(adultNames)
}